stats/opentelemetry/e2e_test.go

```diff
@@ -582,3 +620,868 @@ func pollForWantMetrics(ctx context.Context, t *testing.T, reader *metric.Manual
 
 	return fmt.Errorf("error waiting for metrics %v: %v", wantMetrics, ctx.Err())
 }
+
+// TestMetricsAndTracesOptionEnabled verifies the integration of metrics and traces
+// emitted by the OpenTelemetry instrumentation in a gRPC environment. It sets up a
+// stub server with both metrics and traces enabled, and tests the correct emission
+// of metrics during a Unary RPC and a Streaming RPC. The test ensures that the
+// emitted metrics reflect the operations performed, including the size of the
+// compressed message, and verifies that tracing information is correctly recorded.
+func (s) TestMetricsAndTracesOptionEnabled(t *testing.T) {
```

Comment:
@aranjans may be one more tests to verify metrics and traces are not emitted if they are not enabled?

Reply:
@purnesh42H I have added the test. Please note: I have added the test which verifies the current setup works as expected when traces and metrics are disabled.